starknet_py/net/account/account.py

```diff
@@ -0,0 +1,420 @@
+        block_hash: Optional[Union[Hash, Tag]] = None,
+        block_number: Optional[Union[int, Tag]] = None,
+    ) -> EstimatedFee:
+        """
+        :param tx: Transaction which fee we want to calculate.
+        :param block_hash: a block hash.
+        :param block_number: a block number.
+        :return: Estimated fee
+        """
+        signature = self.signer.sign_transaction(tx)
+        tx = _add_signature_to_transaction(tx, signature)
+
+        return await self._client.estimate_fee(
+            tx=tx,
+            block_hash=block_hash,
+            block_number=block_number,
+        )
+
+    async def get_nonce(self) -> int:
+        """
+        Get the current nonce of the account
+
+        :return: nonce
+        """
+        return await self._client.get_contract_nonce(
+            self.address, block_number="pending"
+        )
+
+    async def get_balance(
+        self, token_address: Optional[AddressRepresentation] = None
+    ) -> int:
+        token_address = token_address or default_token_address_for_network(
+            self._client.net
+        )
+
+        low, high = await self._client.call_contract(
+            Call(
+                to_addr=parse_address(token_address),
+                selector=get_selector_from_name("balanceOf"),
+                calldata=[self.address],
+            ),
+            block_hash="pending",
+        )
+
+        return (high << 128) + low
+
+    async def sign_invoke_transaction(
+        self,
+        calls: Calls,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> Invoke:
+        execute_tx = await self._prepare_invoke_function(calls, max_fee, auto_estimate)
+        signature = self.signer.sign_transaction(execute_tx)
+        return _add_signature_to_transaction(execute_tx, signature)
+
+    async def sign_declare_transaction(
+        self,
+        compiled_contract: str,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> Declare:
+        compiled_contract = create_compiled_contract(
+            compiled_contract=compiled_contract
+        )
+        declare_tx = Declare(
+            contract_class=compiled_contract,
+            sender_address=self.address,
+            max_fee=0,
+            signature=[],
+            nonce=await self.get_nonce(),
+            version=self._version,
+        )
+
+        max_fee = await self._get_max_fee(
+            transaction=declare_tx, max_fee=max_fee, auto_estimate=auto_estimate
+        )
+        declare_tx = _add_max_fee_to_transaction(declare_tx, max_fee)
+        signature = self.signer.sign_transaction(declare_tx)
+        return _add_signature_to_transaction(declare_tx, signature)
+
+    async def sign_deploy_account_transaction(
+        self,
+        class_hash: int,
+        contract_address_salt: int,
+        constructor_calldata: Optional[List[int]] = None,
```

Comment:
I think all arguments could be keyword ones

Reply:
I think it's fine to have arguments that will always be present non-keyword. Perhaps `constructor_calldata` could be changed to keyword-only, keeping the rest?

cc @war-in @drknzz

Reply:
It's completely up to you. I do not have a strong opinion on that

Reply:
I though about this and I think making these to be keyword only is not very ergonomic for the end user.